webmap/src/control/ContextMenuControl.ts

```diff
@@ -20,42 +21,63 @@ export default class ContextMenuControl extends L.Control {
+        if (css !== undefined) {
+            insertCss(css, this._id);
         }
     }
 
     onAdd(): HTMLDivElement {
         this._dom = L.DomUtil.create('div', 'leaflet-control leaflet-control-contextmenu');
         this._dom.dataset.label = this._pl3xmap.settings!.lang.contextMenu.label;
-        if (this._customHtml.enabled) {
-            this._dom.innerHTML = this._customHtml.html;
-        }
         return this._dom;
     }
 
+
+    onRemove(): void {
+        removeCss(this._id);
+    }
+
     private _show(event: L.LeafletMouseEvent): void {
-        L.DomEvent.stopPropagation(event);
-        if (!this._customHtml.enabled) {
-            this._dom.innerHTML = '';
-            this._getItems(event).forEach((item) => {
-                const menuItem = L.DomUtil.create('div', 'leaflet-control-contextmenu-item', this._dom);
-                menuItem.innerHTML = item.label;
-                L.DomEvent.on(menuItem, 'click', (e) => {
-                    L.DomEvent.stopPropagation(e);
-                    item.callback();
-                    this._hide();
-                });
-            });
+        // Ignore right-clicks on controls (https://github.com/JLyne/LiveAtlas/blob/0819cdf2728b49d361f9adfda09ff08311a59337/src/components/map/MapContextMenu.vue#L188-L194)
+        if(event.originalEvent.target && (event.originalEvent.target as HTMLElement).closest('.leaflet-control')) {
+            return;
         }
-        this._dom.style.display = 'block';
-        this._dom.style.left = event.containerPoint.x + 'px';
-        this._dom.style.top = event.containerPoint.y + 'px';
+
+        event.originalEvent.stopImmediatePropagation();
+        event.originalEvent.preventDefault();
+
+        this._dom.style.visibility = 'visible';
+
+        this._dom.innerHTML = '';
+        this._getItems(event).forEach((item) => {
+            const menuItem = L.DomUtil.create('button', 'leaflet-control-contextmenu-item', this._dom);
+            menuItem.innerHTML = item.label;
+            L.DomEvent.on(menuItem, 'click', (e) => {
+                L.DomEvent.stopPropagation(e);
+                item.callback();
+                this._hide();
+            });
+        });
+
+        // Don't position offscreen (https://github.com/JLyne/LiveAtlas/blob/0819cdf2728b49d361f9adfda09ff08311a59337/src/components/map/MapContextMenu.vue#L123-L135)
+        const x = Math.min(
+                window.innerWidth - this._dom.offsetWidth - 10,
+                event.originalEvent.clientX
+            ),
+            y = Math.min(
+                window.innerHeight - this._dom.offsetHeight - 10,
+                event.originalEvent.clientY
+            );
+
+        this._dom.style.display = 'flex';
+        this._dom.style.transform = `translate(${x}px, ${y}px)`;
     }
 
     private _hide(): void {
-        this._dom.style.display = 'none';
+        this._dom.style.visibility = 'hidden';
+        this._dom.style.left = '-1000';
```

Comment:
What's this for?

Reply:
`visibility`: to replace modifying the `display` style
`left`: to move it out of the window for accessibility reasons